File: suites/02_numerical_computation/comparisons/matrix_multiply.py
def matrix_checksum(size: int) -> int:
    s = 0
    for row in range(size):
        for col in range(size):
            cell = 0
            for inner in range(size):
                cell += ((row * inner) + 3) * ((inner * col) + 5)
            s += cell % 104729
    return s

if __name__ == "__main__":
    print(matrix_checksum(64))

File: suites/02_numerical_computation/comparisons/matrix_multiply.go
package main
import "fmt"
func matrixChecksum(size int64) int64 {
	var sum int64
	for row := int64(0); row < size; row++ {
		for col := int64(0); col < size; col++ {
			var cell int64
			for inner := int64(0); inner < size; inner++ {
				cell += ((row * inner) + 3) * ((inner * col) + 5)
			}
			sum += cell % 104729
		}
	}
	return sum
}
func main() { fmt.Println(matrixChecksum(64)) }
